lib/web_ui/lib/src/engine/text/canvas_paragraph.dart

```diff
@@ -169,15 +170,18 @@ class CanvasParagraph implements ui.Paragraph {
         if (text.isEmpty) {
           continue;
         }
-
+        fullText += text;
+      }
+      if (line.fragments.isNotEmpty) {
+        final LayoutFragment firstFragment = line.fragments[0];
         final DomElement spanElement = domDocument.createElement('flt-span');
-        if (fragment.textDirection == ui.TextDirection.rtl) {
+        if (firstFragment.textDirection == ui.TextDirection.rtl) {
           spanElement.setAttribute('dir', 'rtl');
         }
-        applyTextStyleToElement(element: spanElement, style: fragment.style);
-        _positionSpanElement(spanElement, line, fragment);
-
-        spanElement.appendText(text);
+        applyTextStyleToElement(
+            element: spanElement, style: firstFragment.style);
+        _positionSpanElement(spanElement, line, firstFragment);
+        spanElement.appendText(fullText);
```

Comment:
It looks like you are putting the entire line into a single `<span>` element, is that correct?

This won't work. One line of text could contain fragments with different styles. You can't just put all of them in the same `<span>` with the same style.